From 4f189cc165127721d1b13c28e40005004bf77bba Mon Sep 17 00:00:00 2001
From: Udipto Goswami <udipto.goswami@oss.qualcomm.com>
Date: Thu, 19 Mar 2026 01:16:25 +0530
Subject: [PATCH 39/63] ANDROID: usb: xhci: Expose xhci_handle_event_trb() for
 vendor access

Vendor handler requires the xhci_handle_event_trb() in order to process
the individual trbs in event using the event handling logic.

Export the function to make it available in vendor code.

Bug: 494307525
Bug: 487900539
Bug: 497692253
Change-Id: Id63ce6b935aea3e96e6217ad12c214e828cf0c6f
Signed-off-by: Udipto Goswami <udipto.goswami@oss.qualcomm.com>
---
 drivers/usb/host/xhci-ring.c | 3 ++-
 drivers/usb/host/xhci.h      | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index 5cc0fc32546a..ffa66c5648de 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -3028,7 +3028,7 @@ static int handle_tx_event(struct xhci_hcd *xhci,
  * This function handles one OS-owned event on the event ring. It may drop
  * xhci->lock between event processing (e.g. to pass up port status changes).
  */
-static int xhci_handle_event_trb(struct xhci_hcd *xhci, struct xhci_interrupter *ir,
+int xhci_handle_event_trb(struct xhci_hcd *xhci, struct xhci_interrupter *ir,
 				 union xhci_trb *event)
 {
 	u32 trb_type;
@@ -3072,6 +3072,7 @@ static int xhci_handle_event_trb(struct xhci_hcd *xhci, struct xhci_interrupter
 
 	return 0;
 }
+EXPORT_SYMBOL_GPL(xhci_handle_event_trb);
 
 /*
  * Update Event Ring Dequeue Pointer:
diff --git a/drivers/usb/host/xhci.h b/drivers/usb/host/xhci.h
index 86dec9ef5246..81cc7d0f4d2b 100644
--- a/drivers/usb/host/xhci.h
+++ b/drivers/usb/host/xhci.h
@@ -1961,7 +1961,8 @@ void xhci_process_cancelled_tds(struct xhci_virt_ep *ep);
 void xhci_update_erst_dequeue(struct xhci_hcd *xhci,
 			      struct xhci_interrupter *ir,
 			      bool clear_ehb);
-
+int xhci_handle_event_trb(struct xhci_hcd *xhci, struct xhci_interrupter *ir,
+			union xhci_trb *event);
 /* xHCI roothub code */
 void xhci_set_link_state(struct xhci_hcd *xhci, struct xhci_port *port,
 				u32 link_state);
-- 
2.50.1

